### ISSUE TITLE
[@typescript-eslint/require-await] false positive on decorated async methods 

### ISSUE THREAD

**Issue by Extacy** [typescript-eslint/typescript-eslint#4423]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
     "require-await": "off",
     "@typescript-eslint/require-await": "error"
  }
}
```

```TS
function TestDecorator(): MethodDecorator {
    return function () { };
}

class Bruh {
    @TestDecorator()
    public good() {
        return "Hello, World";
    }

    @TestDecorator()
    public async bad() {
        return "Hello, World";
    }
}
```

**Expected Result**

No error
**Actual Result**
```
C:\Users\...\src\main.ts
  15:5  error  Async method 'bad' has no 'await' expression  @typescript-eslint/require-await
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^5.9.0` |
| `@typescript-eslint/parser`        | `^5.9.0` |
| `TypeScript`                       | `4.4.3` |
| `ESLint`                           | `^8.6.0` |
| `node`                             | `16.11.0` |

**Comment by bradzacher** [comment]
What about this, exactly, makes it a false positive?

Just because you add a decorator doesn't make it okay to have Async without an await.

**Comment by Extacy** [comment]
Oh whoops, I misread the details about the rule.

I was under the impression that `require-await` would prevent async functions from being called without using await, but I see it just prevents unnecessary async functions

I think a rule like that would be pretty useful, but thank you anyway 👍🏿

**Comment by bradzacher** [comment]
Check the `no-floating-promise` lint rule which will warn against unhandled promises
